bugrap/src/main/java/org/vaadin/bugrap/ui/columns/FineDateRenderer.java

```diff
@@ -0,0 +1,64 @@
+package org.vaadin.bugrap.ui.columns;
+
+import java.io.ObjectInputStream.GetField;
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.TimeZone;
+
+import com.vaadin.ui.renderers.DateRenderer;
+
+import elemental.json.JsonValue;
+
+public class FineDateRenderer extends DateRenderer{
+
+	private final int[] units = {Calendar.YEAR, Calendar.MONTH, Calendar.DAY_OF_YEAR, Calendar.HOUR_OF_DAY, Calendar.MINUTE, Calendar.SECOND};
+	private final String[] unitTextValues = {"year", "month", "day", "hour", "minute", "second"};
+	
+	
+	@Override
+	public JsonValue encode(Date value) {
+		return super.encode(getFineTextFromDate(value), String.class);
+	}
+	
+	private String getFineTextFromDate(Date value) {
+		StringBuffer presentationText = new StringBuffer();
+		if (value == null)
+			return presentationText.toString();
+		
+		long diffInMillis = (new Date()).getTime() - value.getTime();
+		
+		Calendar diff = Calendar.getInstance();
+		diff.setTimeInMillis(diffInMillis - diff.get(Calendar.ZONE_OFFSET));
```

Comment:
Why do you have to subtract the timezone offset? Maybe some Java api can help you calculating differences between dates (and times).